python/ray/experimental/gpu_object_manager/gpu_object_manager.py

```diff
@@ -105,24 +120,93 @@ def __init__(self):
 
     @property
     def gpu_object_store(self) -> "ray.experimental.GPUObjectStore":
-        with self.gpu_object_store_lock:
+        with self._lock:
```

Comment:
Hmm I think it is clearer to have a separate lock for singletons vs the tensor metadata. Conceptually they are responsible for different state.